tests/test_hosts.py

```diff
@@ -0,0 +1,36 @@
+# -*- coding: utf-8 -*-
+from __future__ import print_function
+from pyloginsight.models import Hosts
+import logging
+import random
+
+
+logger = logging.getLogger(__name__)
+
+
+def test_iterate_over_hosts(connection):
+    hostsobject = Hosts(connection)
+    counter = 0
+    for h in hostsobject:
+        assert hasattr(h, "hostname")
+        assert hasattr(h, "last_received")
+        assert hasattr(h, "source")
+        counter += 1
+    assert counter > 0
+    assert counter == len(hostsobject)
```

Comment:
The pagination code could, if it has a bug, return the same Host object multiple times. Record the seen-hosts in a list or set, and prove there are no duplicates when inserting.

Reply:
I added a test, found that the sorting was all off, and simplified the hosts implementation as a result.